src/hotspot/share/runtime/vframeArray.cpp

```diff
@@ -95,9 +94,16 @@ void vframeArrayElement::fill_in(compiledVFrame* vf, bool realloc_failures) {
         if (monitor->owner_is_scalar_replaced()) {
           dest->set_obj(nullptr);
```

Comment:
It looks like there is an existing bug that allows dest->lock() to be uninitialized here, which could cause problems later on in unpack_on_stack if move_to sees a "neutral" value and tries to inflate.

Reply:
I created https://bugs.openjdk.org/browse/JDK-8331918 so this can be looked at separately.